test/integration/esys-tr-getName.int.c

```diff
@@ -0,0 +1,79 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*******************************************************************************
+ * Copyright 2017-2018, Fraunhofer SIT sponsored by Infineon Technologies AG
+ * All rights reserved.
+ *******************************************************************************/
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#include <stdlib.h>
+#include <stdio.h>
+#include "tss2_mu.h"
+#include "tss2_esys.h"
+
+#include "esys_iutil.h"
+#define LOGMODULE test
+#include "util/log.h"
+#include "util/aux_util.h"
+
+/** This tests the Esys_TR_GetTpmHandle and Esys_TR_GetName functions by
+ *  using a dummy AC handle and validating it against the expected TPM handle
+ *  and TPM name.
+ *
+ * Tested ESYS commands:
+ * - Esys_TR_GetTpmHandle
+ * - Esys_TR_GetName
+ *
+ * @param[in,out] ectx The ESYS_CONTEXT.
+ * @retval EXIT_FAILURE
+ * @retval EXIT_SUCCESS
+ */
+
+int
+test_esys_tr_getName(ESYS_CONTEXT * ectx)
+{
+    TSS2_RC r;
+
+    ESYS_TR ac = ESYS_TR_RH_AC(1); /* pick second AC */
+    TPM2_HANDLE tpmHandle = ESYS_TR_NONE;
+
+    r = Esys_TR_GetTpmHandle(ectx, ac, &tpmHandle);
+    goto_if_error(r, "TR get TPM handle", error);
+
+    if (tpmHandle != TPM2_RH_ACT_1) {
+        LOG_ERROR("Handles mismatch");
+        goto error;
+    }
+
+    TPM2B_NAME name1, *name2;
+    size_t offset = 0;
+
+    r = Tss2_MU_TPM2_HANDLE_Marshal(TPM2_RH_ACT_1, &name1.name[0],
```

Comment:
TPM2_NV_AC_FIRST not `_ACT_`

Clever to take the handle and marshal it to a name buffer.